### PR TITLE
351 family list from coveo

### LINKED ISSUE #351: [Product Category] Family List (based on Coveo)

### PR DESCRIPTION
**Pull request by rgravitvl**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #351

Test URLs:
- Before: https://main--danaher-ls-aem--hlxsites.hlx.page/
- After: https://351-family-list-from-coveo--danaher-ls-aem--hlxsites.hlx.page/us/en/products/centrifuges/analytical-ultracentrifuges
- After: https://351-family-list-from-coveo--danaher-ls-aem--hlxsites.hlx.page/us/en/products/microscopes/light-microscopes/inverted-light-microscopes

### CODE REVIEW

File: blocks/category-family/category-family.js [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    organizationId: window.DanaherConfig.searchOrg,
+    organizationEndpoints: await categorySearchInterface
+      .getOrganizationEndpoints(window.DanaherConfig.searchOrg),
+  });
+
+  const isInternal = typeof getCookie('exclude-from-analytics') !== 'undefined';
+  const { engine } = categorySearchInterface;
+  engine.dispatch(loadContextActions(engine).setContext({
+    categories: category,
+    host: 'stage.lifesciences.danaher.com', // window.location.host,
```
Comment (davidzochowski) [inline blocks/category-family/category-family.js:222]: Let's set this to window.location.host before we forget. I can setup a rule in Coveo to serve stage families if the host is *.hlx.page

File: blocks/category-family/category-family.js [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+const isOTEnabled = () => {
+  const otCookie = getCookie('OptanonConsent');
+  if (typeof otCookie === 'string') {
+    return otCookie.includes('C0002:1');
+  }
+  return true;
+};
+
+export default async function decorate(block) {
+  const category = window.location.pathname.slice(16).split('.').at(0).replaceAll('/', '|');
```
Comment (davidzochowski) [inline blocks/category-family/category-family.js:204]: We need to take a second look into this and see if we can figure out a better way to get the category path. When we load the page through google's webcache URL, this caused the category info component we had before to fetch the incorrect category information from Coveo.
@davenichols-DHLS

File: blocks/category-family/category-family.js [hunk 3]
```diff
@@ -0,0 +1,225 @@
+// eslint-disable-next-line import/no-unresolved
+import { loadContextActions } from 'https://static.cloud.coveo.com/headless/v2/headless.esm.js';
+import { loadScript } from '../../scripts/lib-franklin.js';
+
+const categoryFamily = `
+    <atomic-search-interface class="category-search" localization-compatibility-version="v4"
+        search-hub="DanaherCategoryProductListing" pipeline="Danaher Category Product Listing"
```
Comment (davidzochowski) [inline blocks/category-family/category-family.js:7]: I'm doing some renaming/cleanup within Coveo to prepare for the OpCo migrations. As part of it, I have renamed the pipeline and the searchhub to:
    "pipeline": "Danaher LifeSciences Category Product Listing"
    "searchHub": "DanaherLifeSciencesCategoryProductListing"
Would you please update the values in the html?

Comment (rgravitvl) [reply blocks/category-family/category-family.js:7]: Done

**Review by davidzochowski — APPROVED** [review]

**Pull request merged**